### PR TITLE
add new keyboard 'soda/cherish'

### PR DESCRIPTION
**Pull request by gezhaoyou**
## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/soda/cherish/config.h [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+*/
+
+#pragma once
+
+/* USB Device descriptor parameter */
+#define VENDOR_ID       0xEB50 // AP for Cherish-75
+#define PRODUCT_ID      0xEB52 // AR for Cherish-75
+#define DEVICE_VER      0x0001 // Revision pre-Alpha
+#define MANUFACTURER    gezhaoyou
+#define PRODUCT         Cherish-75
```
Comment (drashna) [inline keyboards/soda/cherish/config.h:25]:
```suggestion
```

File: keyboards/soda/cherish/config.h [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+//#define NO_PRINT
+
+/* disable action features */
+//#define NO_ACTION_LAYER
+//#define NO_ACTION_TAPPING
+//#define NO_ACTION_ONESHOT
+
+#define RGB_DI_PIN A13
+#ifdef RGB_DI_PIN
+#define RGBLIGHT_ANIMATIONS
```
Comment (drashna) [inline keyboards/soda/cherish/config.h:72]: Could you explicitly define which animations are enabled? 
```suggestion
#define RGBLIGHT_EFFECT_BREATHING
#define RGBLIGHT_EFFECT_RAINBOW_MOOD
#define RGBLIGHT_EFFECT_RAINBOW_SWIRL
#define RGBLIGHT_EFFECT_SNAKE
#define RGBLIGHT_EFFECT_KNIGHT
#define RGBLIGHT_EFFECT_CHRISTMAS
#define RGBLIGHT_EFFECT_STATIC_GRADIENT
#define RGBLIGHT_EFFECT_RGB_TEST
#define RGBLIGHT_EFFECT_ALTERNATING
#define RGBLIGHT_EFFECT_TWINKLE
```

File: keyboards/soda/cherish/info.json [hunk 3]
```diff
@@ -0,0 +1,97 @@
+{
+    "keyboard_name": "75% ANSI layout",
+    "url": "",
+    "maintainer": "qmk",
+    "layouts": {
```
Comment (drashna) [inline keyboards/soda/cherish/info.json:5]:
```suggestion
    "manufacturer": "gezhaoyou",
    "usb": {
        "vid": "0xEB50",
        "pid": "0xEB52",
        "device_version": "0.0.1"
    },
    "layouts": {
```

File: keyboards/soda/cherish/info.json [hunk 4]
```diff
@@ -0,0 +1,97 @@
+{
+    "keyboard_name": "75% ANSI layout",
```
Comment (drashna) [inline keyboards/soda/cherish/info.json:2]:
```suggestion
    "keyboard_name": "Cherish-75",
```

File: keyboards/soda/cherish/rules.mk [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+CONSOLE_ENABLE = yes        # Console for debug
+COMMAND_ENABLE = yes        # Commands for debug and configuration
+NKRO_ENABLE = yes           # Enable N-Key Rollover
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = yes        # Enable keyboard RGB underglow
+AUDIO_ENABLE = no           # Audio output
+
+# Enter lower-power sleep mode when on the ChibiOS idle thread
+OPT_DEFS += -DCORTEX_ENABLE_WFI_IDLE=TRUE
+VIA_ENABLE = yes
```
Comment (drashna) [inline keyboards/soda/cherish/rules.mk:22]: VIA should be enabled at the keymap level, not the keyboard level.
```suggestion
```

**Review by drashna — CHANGES_REQUESTED** [review]

**Review by fauxpark — APPROVED** [review]

**Review by drashna — APPROVED** [review]

**Pull request merged**